crates/apollo-router-core/src/query_planner/model.rs

```diff
@@ -386,43 +386,46 @@ async fn fetch_node<'a>(
             Some(Response {
                 data, mut errors, ..
             }) => {
-                if let Some(entities) = data.get("_entities") {
-                    tracing::trace!(
-                        "Received entities: {}",
-                        serde_json::to_string(entities).unwrap(),
-                    );
-                    if let Some(array) = entities.as_array() {
-                        let mut response = response
-                            .lock()
-                            .instrument(tracing::trace_span!("response_lock_wait"))
-                            .await;
-
-                        let span = tracing::trace_span!("response_insert");
-                        let _guard = span.enter();
-                        for (i, entity) in array.iter().enumerate() {
-                            response.insert_data(
-                                &current_dir.join(Path::from(i.to_string())),
-                                entity,
-                            )?;
+                if let Value::Object(mut map) = data {
```

Comment:
https://docs.serde.rs/serde_json/value/enum.Value.html#method.get allows you to write `data.get("_entities")`.

I suspect you went with a double if so you can use `map.remove`, but I had a bit of a hard time making sure the logic remained the same, possibly with one more level of nesting 😅, especially the missyng key `entities` error bit at line 420

Reply:
I added a comment for that. Unfortunately, serde_josn's API does not allow better patterns